core/src/main/java/org/infinispan/interceptors/VersionedEntryWrappingInterceptor.java

```diff
@@ -41,33 +41,47 @@ public void initialize(VersionGenerator versionGenerator) {
 
    @Override
    public Object visitPrepareCommand(TxInvocationContext ctx, PrepareCommand command) throws Throwable {
+      VersionedPrepareCommand versionedPrepareCommand = (VersionedPrepareCommand) command;
```

Comment:
Yeah, I have another PR coming up that's more focused on the entry wrapping, so I could move it there.